src/Protocol/Http/Authenticator/JwtAuthenticator.php

```diff
@@ -40,14 +55,37 @@ public function __construct(JwtInterface $jwt)
         $this->jwt = $jwt;
     }
 
+    /**
+     * Get jws lifetime
+     *
+     * @return integer
+    */
+    public function getJwsLifetime() : int
+    {
+        return $this->jwsLifetime;
+    }
+
+    /**
+     * Set jws lifetime
+     *
+     * @param integer $jwsLifetime
+     */
+    public function setJwsLifetime(int  $jwsLifetime)
+    {
+        $this->jwsLifetime = abs($jwsLifetime);//ignore sign
+    }
+
     /**
      * {@inheritdoc}
      */
     public function authenticate(Request $request): Request
     {
-        $jws = $this->createJwsContent();
+        if (empty($this->jws) || $this->jwsCreatedAt < time() - $this->jwsLifetime) {
```

Comment:
Please remove the `empty` literal. We know, what the `$this->jws` exists.

Note: Many developers (and i) declare the `empty` or `isset` on for can not existence variables: